util/detectOS.ts

```diff
@@ -0,0 +1,39 @@
+// From https://github.com/nodejs/nodejs.dev/blob/main/src/util/detectOS.ts
+// eslint-disable-next-line no-shadow
+export enum UserOS {
+  MAC = 'MAC',
+  WIN = 'WIN',
+  UNIX = 'UNIX',
+  LINUX = 'LINUX',
+  MOBILE = 'MOBILE',
+  UNKNOWN = 'UNKNOWN',
+}
+
+export function detectOS(): UserOS {
```

Comment:
I'd rather use arrow functions here if possible, as I guess we use arrow functions across the codebase?